toolchain/base/value_store.h

```diff
@@ -257,10 +258,11 @@ class SharedValueStores : public Yaml::Printable<SharedValueStores> {
   auto ints() const -> const ValueStore<IntId>& { return ints_; }
   auto reals() -> ValueStore<RealId>& { return reals_; }
   auto reals() const -> const ValueStore<RealId>& { return reals_; }
-  auto string_literals() -> StringStoreWrapper<StringLiteralId>& {
+  auto string_literals() -> StringStoreWrapper<StringLiteralValueId>& {
     return string_literals_;
   }
-  auto string_literals() const -> const StringStoreWrapper<StringLiteralId>& {
+  auto string_literals() const
```

Comment:
I'd make a similar comment here, for StringLiteralValue. Should IdentifierId also be IdentifierValueId, etc?

Reply:
My thinking is that a "string literal" is a kind of token and a kind of expression, and so we have `Lex::TokenKind::StringLiteral`, `Parse::NodeKind::StringLiteral` and `SemIR::StringLiteral`, and we're about to gain a `Parse::StringLiteral` and a `Parse::StringLiteralId`, all representing that token / expression. What we're representing here is the *value* of the string literal expression, not the string literal expression / token itself.

On the other hand, we use "identifier" to refer only to the lexical element; the parse node is an `IdentifierName` or similar, and the semantic node is a `NameRef` or similar, so using `IdentifierId` to refer to the identity of the identifier seems reasonable. And I think the "value" of an identifier could be interpreted as meaning the value of the entity named by that identifier, rather than the name itself.

Reply:
To the extent that "Identifier" is the token too, I'm a bit confused why that conflict isn't an issue.

To a point though, we also do IntegerId for the IntegerLiteral token. Maybe this should just be StringId for a StringLiteral token?